packages/shared/src/components/Markdown.tsx

```diff
@@ -1,12 +1,83 @@
-import React, { ReactElement } from 'react';
+import React, { ReactElement, useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
+import ReactDOM from 'react-dom';
+import { tippy } from '@tippyjs/react';
 import { sanitize } from 'dompurify';
 import styles from './markdown.module.css';
+import { ProfileTooltipContent } from './profile/ProfileTooltipContent';
+import { profileTooltipClasses } from './profile/ProfileTooltip';
+import { useProfileTooltip } from '../hooks/useProfileTooltip';
+import { getProfile, PublicProfile } from '../lib/user';
 
-export default function Markdown({
-  content,
-}: {
+const classes = Object.values(profileTooltipClasses).join(' ');
+
+interface MarkdownProps {
   content: string;
-}): ReactElement {
+}
+
+type TippyInstance = ReturnType<typeof tippy>[0];
+
+const getUserPermalink = (username: string) =>
+  `${process.env.NEXT_PUBLIC_WEBAPP_URL}${username}`;
+
+export default function Markdown({ content }: MarkdownProps): ReactElement {
+  const [username, setUsername] = useState('');
+  const [instance, setInstance] = useState<TippyInstance>();
+  const queryKey = ['profile', username];
+  const { data: fetchedProfile } = useQuery<PublicProfile>(
+    queryKey,
+    () => getProfile(username),
+    { enabled: !!username },
+  );
+  const { fetchInfo, data } = useProfileTooltip(fetchedProfile?.id);
```

Comment:
Here we do double fetch. We run `getProfile` and inside `useProfileTooltip` we ask for the profile tooltip info.
Why can't we do only one? Maybe instead of filling the username in `data-mention-username` we should fill the user id. It will save a call. What do you think?